### DISCUSSION TITLE
Need some way to display the save status of the current page

### DISCUSSION THREAD

**Discussion by PiorGit** [zim-desktop-wiki/zim-desktop-wiki#2558] [Ideas]
Hello,
I am a new user, loving the software so far.

One thing I find sorely missing is the lack of some sort of visual confirmation that the current page has been saved  to disc.

Of course it seems that everything gets saved all the time, hence one could argue that an indicator isn't necessary ; but still, having some sort of confirmation would be great. For instance Google Docs does this with the mention "Saved To Drive", and it could be as simple as merely showing an asterix at the beginning of the page name, like notepad and notepad++ do.

In practice the current UX makes me compulsively hit "ctrl-S" quite often even though it is probably not needed - meaning that the lack of an indicator actually makes me perform some unneeded actions. Furthermore, not knowing what is going on under the hood makes me feel like some day I may believe that my changes are saved even though they are not (for instance if the drive is busy).

**Comment by introt** [comment]
Hi! This'd make a fine addition to the Window Title Editor plugin - a pretty simple one, too: just need to hook to the `modified-changed` signal to add the `*` to the title and `stored-page` to remove it - the Zeitgeist plugin can be used as a reference. Anyone up to give it a go?

**Comment by PiorGit** [comment]
Hello ! Well that's great to hear, I'll definitely be looking forward to it.
At this time Is there some way for the user to check the save status of a page/journal entry ? I am slowly getting used to not thinking about it too much, but it still feels odd at times ...

**Comment by jaap-karssenberg** [comment]
There is no direct status. But automatic save happens every 10 sec out so
in background and as foreground action e.g. when switching pages. If it
fails you get a pop-up so you can save elsewhere. Would hope this to be
sufficient catch all for any mess up.

I always advise to enable version control on notebooks. Because it allows
recovery of you make mistakes like the famous select-all-del combo
inadvertently.

Regards

Jaap

Op do 1 feb. 2024 11:30 schreef P.O. ***@***.***>:

**Comment by PiorGit** [comment]
Hello Jaap, thank you for chiming in.

Knowing that switching pages performs a save is already great to know - and indeed I can see it happen, as selecting/clicking on different pages in the index does cause a very short but noticeable pause every now and then.

Besides that I'll take the advice of running version control on a note-taking software ... as a symptom of a design issue and need for some well-designed auto backup, as opposed to a solution for the lack of it :D For reference see the excellent Blender implementations of both autosave and session recovery.

What is the key combo you mention ? I personally tend to disable any default keyboard shortcut and set up my own when I start to rely heavily on a given software in part for exactly that kind of scenario, but I'd like to know more about this one in particular in case I haven't gotten rid of it yet or in case it is hardcoded.
